questions/questions.js

```diff
@@ -23,18 +23,95 @@ function getStartingQuestions() {
+const skillLevels = [1, 2, 3, 4, 5];
 
-// const skillLevel = ["Elementary", "Intermediate", "Advanced", "Proficient"];
+const chosenSkillLevelAnswers = {
+  math: {
+    chosenAnswer: null,
+  },
+  english: {
+    chosenAnswer: null,
+  },
+  javascript: {
+    chosenAnswer: null,
+  },
+};
 
-// const radioButton = document.createElement("input");
-// radioButton.setAttribute("type", "radio");
-// radioButton.value = optionValue;
+window.onload = () => {
+  renderStartingQuestions(startingQuestions);
+  renderSkillsLevelQuiz(skillLevels, subjects);
+};
 
-// const label = document.createElement("label");
-// label.innerHTML = optionValue;
+const formElem = document.querySelector("form");
+const skillBox = document.createElement("ul");
+skillBox.classList.add("choices");
 
-// radio buttons
-//Math
-//English
-//JS
+formElem.addEventListener("submit", (event) => {
+  event.preventDefault();
+});
+
+// handleMathButtonClick() also for JS and English
+// higher-order function name as a parameter
+// makeButtonClickHandler()
+
+function makeButtonClickHandler() {
+  function handleMathButtonClick(event, subjects) {
+    for (let subject of subjects) {
+      subject = subject.toLowerCase();
+      if (subject === event.target.name) {
+        chosenSkillLevelAnswers[subject]["chosenAnswer"] = event.target.value;
+      }
+    }
+    return chosenSkillLevelAnswers;
+  }
+  return handleMathButtonClick(event, subjects);
+}
+
+// function handleMathButtonClick(event) {
+//   chosenSkillLevelAnswers.math.chosenAnswer = event.target.value; // or with Number
+// }
+
+// function handleEnglishButtonClick(event) {
+//   chosenSkillLevelAnswers.english.chosenAnswer = event.target.value; // or with Number
+// }
+
+// function handleJavascriptButtonClick(event) {
+//   chosenSkillLevelAnswers.javascript.chosenAnswer = event.target.value; // or with Number
+// }
+
+function renderSkillsLevelQuiz(skillLevels, subjects) {
+  for (let subject of subjects) {
+    const subjectElem = document.createElement("span");
+    subjectElem.append(subject);
+    skillBox.append(subjectElem);
+    for (let skillLevel of skillLevels) {
+      const skillLevelElem = document.createElement("li");
+      const radioBtn = document.createElement("input");
+      radioBtn.setAttribute("type", "radio");
+      radioBtn.name = subject.toLowerCase();
+      radioBtn.value = Number(skillLevel);
+      const label = document.createElement("label");
+      label.append(radioBtn);
+      // if (radioBtn.name === "math") {
+      //   radioBtn.addEventListener("change", handleMathButtonClick);
+      // }
+
+      // if (radioBtn.name === "english") {
+      //   radioBtn.addEventListener("change", handleEnglishButtonClick);
+      // }
+
+      // if (radioBtn.name === "javascript") {
+      //   radioBtn.addEventListener("change", handleJavascriptButtonClick);
+      // }
+
+      radioBtn.addEventListener("change", makeButtonClickHandler);
```

Comment:
you need to call makeButtonClickHandler with the subject of this current button, then use the returned function as an event handler.